packages/paste-website/src/pages/primitives/combobox-primitive/index.mdx

```diff
@@ -0,0 +1,293 @@
+
+The purpose of providing these unstyled primitives is to cater for instances when the styled Combobox provided
+by Paste, doesn't meet the requirements needed to solve a unique or individual customer problem. At that point
+you are welcome to fallback to this functional primitive to roll your own styled combobox whilst still providing
+a functional and accessible experience to your customers.
+
+This primitive should be used to compose all custom comboboxes to ensure accessibility and upgrade paths.
+
+<Callout variant="warning">
+  <CalloutTitle>Warning</CalloutTitle>
+  <CalloutText>
+    We strongly suggest that all components built on top of this primitive get reviewed by the Design Systems team and
+    goes through the UX Review process to ensure an excellent experience for our customers.
+  </CalloutText>
+</Callout>
+
+## Usage Guide
+
+This package is a wrapper around the [`Downshift` package](https://downshift.netlify.com).
+Our wrapper currently only exposes the useSelect and useCombobox hooks, but renamed for Paste.
+The API's for both hooks are otherwise identical. The reason we chose to just expose the hooks
+is that we feel these are the most flexible ways of consuming downshift and better fit
+our chosen styling model.
+
+If you’re wondering why we wrapped that package into our own, we reasoned that it would be
+best for our consumers' developer experience. For example:
+
+- If we want to migrate the underlying nuts and bolts in the future, Twilio products
+  that depend on this primitive would need to replace all occurrences of `import … from ‘x-package’`
+  to `import … from ‘@some-new/package’`. By wrapping it in `@twilio-paste/x-primitive`, this
+  refactor can be avoided. The only change would be a version bump in the package.json file for the primitive.
+- We can more strictly enforce semver and backwards compatibility than some of our dependencies.
+- We can control when to provide an update and which versions we whitelist, to help reduce potential
+  bugs our consumers may face.
+- We can control which APIs we expose. For example, we may chose to enable or disable usage of
+  certain undocumented APIs.
+
+### Installation
+
+This package is available individually or as part of `@twilio-paste/core`.
+
+```
+yarn add @twilio-paste/combobox-primitive - or - yarn add @twilio-paste/core
+```
+
+### Examples
+
+#### Basic Combobox
+
+A basic Combobox is a direct replacement for the native HTML select element. It should function in roughly
+the same way, the difference being in the ability to style everything about it; the input trigger, option list
+and option content.
+
+Contrary to the Downshift documentation and example, the basic Combobox should have an HTML input element as
+the trigger, and not an HTML button. Use an input element and set the role of combobox in the `getToggleButtonProps`
+getter or directly on the element itself. You can see this demonstrated below:
```

Comment:
Might this be something we do directly in the primitive before re-exposing it to consumers, to prevent misuse?

Reply:
Because the primitive is just the hook, I don't think we want to lock it down to a particular composition. Guidance might be our best path forward here.